File: master-goods-android/core/model/src/main/java/com/zhihuiji/core/model/ReportModels.kt
package com.zhihuiji.core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class SalesSummaryReportDto(
    @SerialName("start_at") val startAt: Long,
    @SerialName("end_at") val endAt: Long,
    @SerialName("total_sales_amount") val totalSalesAmount: Double = 0.0,
    @SerialName("total_paid_amount") val totalPaidAmount: Double = 0.0,
    @SerialName("total_refund_amount") val totalRefundAmount: Double = 0.0,
    @SerialName("total_unpaid_amount") val totalUnpaidAmount: Double = 0.0,
    @SerialName("total_order_count") val totalOrderCount: Int = 0,
)

@Serializable
data class ProfitSummaryReportDto(
    @SerialName("start_at") val startAt: Long,
    @SerialName("end_at") val endAt: Long,
    @SerialName("estimated_cost_amount") val estimatedCostAmount: Double = 0.0,
    @SerialName("estimated_profit_amount") val estimatedProfitAmount: Double = 0.0,
    @SerialName("estimated_profit_rate") val estimatedProfitRate: Double = 0.0,
)

@Serializable
data class RefundRecordReportDto(
    @SerialName("payment_id") val paymentId: Long,
    @SerialName("order_id") val orderId: Long,
    @SerialName("order_no") val orderNo: String,
    @SerialName("customer_name") val customerName: String,
    @SerialName("refund_amount") val refundAmount: Double,
    val method: Int = 0,
    @SerialName("reference_no") val referenceNo: String? = null,
    @SerialName("created_at") val createdAt: Long,
)

@Serializable
data class StockOutRecordReportDto(
    @SerialName("order_id") val orderId: Long,
    @SerialName("order_no") val orderNo: String,
    @SerialName("customer_id") val customerId: Long? = null,
    @SerialName("customer_name") val customerName: String? = null,
    @SerialName("product_id") val productId: Long,
    @SerialName("product_code") val productCode: String,
    @SerialName("product_name") val productName: String,
    val quantity: Double,
    @SerialName("unit_price") val unitPrice: Double = 0.0,
    val amount: Double,
    @SerialName("item_created_at") val itemCreatedAt: Long,
    @SerialName("order_created_at") val orderCreatedAt: Long,
)

@Serializable
data class TopSellingProductReportDto(
    @SerialName("product_id") val productId: Long,
    @SerialName("product_code") val productCode: String,
    @SerialName("product_name") val productName: String,
    @SerialName("total_quantity") val totalQuantity: Double,
    @SerialName("total_amount") val totalAmount: Double,
)

@Serializable
data class ProfitByProductReportDto(
    @SerialName("product_id") val productId: Long,
    @SerialName("product_code") val productCode: String,
    @SerialName("product_name") val productName: String,
    @SerialName("total_sales_amount") val totalSalesAmount: Double,
    @SerialName("total_cost_amount") val totalCostAmount: Double,
    @SerialName("total_profit_amount") val totalProfitAmount: Double,
    @SerialName("profit_rate") val profitRate: Double = 0.0,
)

@Serializable
data class ProfitByCustomerReportDto(
    @SerialName("customer_id") val customerId: Long? = null,
    @SerialName("customer_name") val customerName: String,
    @SerialName("total_sales_amount") val totalSalesAmount: Double,
    @SerialName("total_cost_amount") val totalCostAmount: Double,
    @SerialName("total_profit_amount") val totalProfitAmount: Double,
    @SerialName("profit_rate") val profitRate: Double = 0.0,
)

@Serializable
data class InventoryFlowRecordDto(
    @SerialName("order_id") val orderId: Long,
    @SerialName("order_no") val orderNo: String,
    @SerialName("product_id") val productId: Long,
    @SerialName("product_code") val productCode: String,
    @SerialName("product_name") val productName: String,
    val quantity: Double,
    @SerialName("flow_type") val flowType: Int,
    @SerialName("flow_time") val flowTime: Long,
    @SerialName("customer_name") val customerName: String? = null,
    @SerialName("source_type") val sourceType: Int,
    @SerialName("source_label") val sourceLabel: String? = null,
    @SerialName("adjust_reason") val adjustReason: String? = null,
    @SerialName("operator_name") val operatorName: String? = null,
)

@Serializable
data class CustomerSalesReportDto(
    @SerialName("customer_id") val customerId: Long? = null,
    @SerialName("customer_name") val customerName: String,
    @SerialName("total_orders") val totalOrders: Int,
    @SerialName("total_amount") val totalAmount: Double,
)

@Serializable
data class CustomerReceivableReportDto(
    @SerialName("customer_id") val customerId: Long,
    @SerialName("customer_name") val customerName: String,
    val phone: String,
    val balance: Double,
)

@Serializable
data class LowStockProductReportDto(
    @SerialName("product_id") val productId: Long,
    @SerialName("product_code") val productCode: String,
    @SerialName("product_name") val productName: String,
    val stock: Double,
    @SerialName("safe_stock") val safeStock: Double,
)

@Serializable
data class ReconciliationSummaryReportDto(
    @SerialName("start_at") val startAt: Long,
    @SerialName("end_at") val endAt: Long,
    @SerialName("total_receivable_amount") val totalReceivableAmount: Double = 0.0,
    @SerialName("total_payable_amount") val totalPayableAmount: Double = 0.0,
    @SerialName("total_received_amount") val totalReceivedAmount: Double = 0.0,
    @SerialName("total_paid_amount") val totalPaidAmount: Double = 0.0,
    @SerialName("net_cash_flow") val netCashFlow: Double = 0.0,
)

File: frontend/ios/ZhihuijiIOS/Core/Models/ReportModels.swift
import Foundation

struct KPIBlock: Identifiable, Codable, Equatable {
    let id: String
    let title: String
    let value: String
    let subtitle: String
}

struct SalesSummaryReport: Codable, Equatable {
    let startAt: Int64
    let endAt: Int64
    let totalSalesAmount: Double
    let totalPaidAmount: Double
    let totalRefundAmount: Double
    let totalUnpaidAmount: Double
    let totalOrderCount: Int
}

struct SalesTrendPoint: Identifiable, Codable, Equatable {
    var id: String { "\(startAt)-\(endAt)" }
    let startAt: Int64
    let endAt: Int64
    let totalSalesAmount: Double
    let totalOrderCount: Int
}

struct ProfitSummaryReport: Codable, Equatable {
    let startAt: Int64
    let endAt: Int64
    let estimatedCostAmount: Double
    let estimatedProfitAmount: Double
    let estimatedProfitRate: Double
}

struct RefundRecordReport: Identifiable, Codable, Equatable {
    var id: String { "\(paymentId)" }
    let paymentId: EntityID
    let orderId: EntityID
    let orderNo: String
    let customerName: String?
    let refundAmount: Double
    let method: Int
    let referenceNo: String?
    let createdAt: Int64
}

struct TopSellingProductReport: Identifiable, Codable, Equatable {
    let productId: EntityID
    let productCode: String
    let productName: String
    let totalQuantity: Double
    let totalAmount: Double

    var id: EntityID { productId }
}

struct ProfitByProductReport: Identifiable, Codable, Equatable {
    let productId: EntityID
    let productCode: String
    let productName: String
    let totalSalesAmount: Double
    let totalCostAmount: Double
    let totalProfitAmount: Double
    let profitRate: Double

    var id: EntityID { productId }
}

struct ProfitByCustomerReport: Identifiable, Codable, Equatable {
    let customerId: EntityID?
    let customerName: String
    let totalSalesAmount: Double
    let totalCostAmount: Double
    let totalProfitAmount: Double
    let profitRate: Double

    var id: String { customerId?.rawValue ?? customerName }
}

struct CustomerSalesReport: Identifiable, Codable, Equatable {
    let customerId: EntityID?
    let customerName: String
    let totalOrders: Int
    let totalAmount: Double

    var id: String { customerId?.rawValue ?? customerName }
}

struct CustomerReceivableReport: Identifiable, Codable, Equatable {
    let customerId: EntityID
    let customerName: String
    let phone: String?
    let balance: Double

    var id: EntityID { customerId }
}

struct LowStockProductReport: Identifiable, Codable, Equatable {
    let productId: EntityID
    let productCode: String
    let productName: String
    let stock: Double
    let safeStock: Double

    var id: EntityID { productId }
}

struct StockOutRecordReport: Identifiable, Codable, Equatable {
    let orderId: EntityID
    let orderNo: String
    let customerId: EntityID?
    let customerName: String?
    let productId: EntityID
    let productCode: String
    let productName: String
    let quantity: Double
    let unitPrice: Double
    let amount: Double
    let itemCreatedAt: Int64
    let orderCreatedAt: Int64

    var id: String { "\(orderId.rawValue)-\(productId.rawValue)-\(itemCreatedAt)" }
}

struct InventoryFlowRecordReport: Identifiable, Codable, Equatable {
    let orderId: EntityID
    let orderNo: String
    let productId: EntityID
    let productCode: String
    let productName: String
    let quantity: Double
    let flowType: Int
    let flowTime: Int64
    let customerName: String?
    let sourceType: Int
    let sourceLabel: String?
    let adjustReason: String?
    let operatorName: String?

    var id: String { "\(orderId.rawValue)-\(productId.rawValue)-\(flowTime)" }
}

struct ReconciliationSummaryReport: Codable, Equatable {
    let startAt: Int64
    let endAt: Int64
    let totalReceivableAmount: Double
    let totalPayableAmount: Double
    let totalReceivableCustomerCount: Int64
    let totalPayableSupplierCount: Int64
    let totalReceivedAmount: Double
    let totalPaidAmount: Double
    let netCashFlow: Double
}

struct CashflowSummaryReport: Codable, Equatable {
    let startAt: Int64
    let endAt: Int64
    let totalIncomeAmount: Double
    let totalExpenseAmount: Double
    let netCashFlow: Double
    let totalRecordCount: Int64
}
